### DISCUSSION TITLE
Texture map is not working

### DISCUSSION THREAD

**Discussion by geralapucha** [bnpr/Malt#72] [Q&A] [answered]
Hello
I have tried to export zbrush models to Blender and render them with Malt using its Shaders, these models have Uvws and Texture Maps for the polypaint, but when applying the basic mesh, it removes the texture map.

Is there a way to make the shaders keep the polypaint?

I am doing something wrong?

Send help: c

**Comment by pragma37** [comment, marked as answer]
Hi!  Sorry for the late reply.
I converted the issue to a discussion since this is not really a bug report.

The *basic.mesh.glsl* example shader doesn't have texture parameters since it serves as the most basic example.
The *basic_texture.mesh.glsl* example shows how to use textures in shaders.

**Reply by geralapucha** [reply]
Hi! Sorry for the late reply :D

So, if i want to use the basic_texture.mesh.glsl and the basic_line_mesh.glsl , both of them, how can i do that?

Sorry, im new with this type of render

**Reply by pragma37** [reply]
Just take a look at the code of both shaders and try to combine them.

It shouldn't be too hard even if you don't have any programming experience.
[This tutorial](https://malt3d.com/#/docs/From-Nodes-To-Code/From-Nodes-To-Code) can help you to understand the basics.

If you find it too difficult I can make the shader for you, but try to do it yourself first.
